compiler/ballerina-lang/src/main/java/io/ballerina/projects/internal/ManifestBuilder.java

```diff
@@ -257,49 +255,53 @@ private PackageDescriptor getPackageDescriptor(TomlTableNode tomlTableNode) {
+                + defaultVersion().value().toString() + "'";
+
         if (tomlTableNode.entries().isEmpty()) {
-            reportDiagnostic(MISSING_PKG_INFO_ERROR,
+            reportDiagnostic(errorMessage,
                     ProjectDiagnosticErrorCode.MISSING_PKG_INFO_IN_BALLERINA_TOML.diagnosticId(),
                     DiagnosticSeverity.WARNING, tomlTableNode.location());
             return PackageDescriptor.from(defaultOrg(), defaultName(this.projectPath), defaultVersion());
         }
 
         TopLevelNode topLevelPkgNode = tomlTableNode.entries().get(PACKAGE);
         if (topLevelPkgNode == null || topLevelPkgNode.kind() != TomlType.TABLE) {
-            reportDiagnostic(MISSING_PKG_INFO_ERROR,
+            reportDiagnostic(errorMessage,
                     ProjectDiagnosticErrorCode.MISSING_PKG_INFO_IN_BALLERINA_TOML.diagnosticId(),
                     DiagnosticSeverity.WARNING, tomlTableNode.location());
             return PackageDescriptor.from(defaultOrg(), defaultName(this.projectPath), defaultVersion());
         }
 
         TomlTableNode pkgNode = (TomlTableNode) topLevelPkgNode;
         if (pkgNode.entries().isEmpty()) {
-            reportDiagnostic(MISSING_PKG_INFO_ERROR,
+            reportDiagnostic(errorMessage,
                     ProjectDiagnosticErrorCode.MISSING_PKG_INFO_IN_BALLERINA_TOML.diagnosticId(),
                     DiagnosticSeverity.WARNING, pkgNode.location());
             return PackageDescriptor.from(defaultOrg(), defaultName(this.projectPath), defaultVersion());
         }
 
         org = getStringValueFromTomlTableNode(pkgNode, "org");
         if (org == null) {
-            reportDiagnostic("missing 'org' under [package]",
+            org = defaultOrg().value();
+            reportDiagnostic("missing 'org' under [package], value derived as '" + org + "'",
                     ProjectDiagnosticErrorCode.MISSING_PKG_INFO_IN_BALLERINA_TOML.diagnosticId(),
                     DiagnosticSeverity.WARNING, pkgNode.location());
-            org = defaultOrg().value();
         }
         name = getStringValueFromTomlTableNode(pkgNode, "name");
         if (name == null) {
-            reportDiagnostic("missing 'name' under [package]",
+            name = defaultName(this.projectPath).value();
+            reportDiagnostic("missing 'name' under [package], value derived as '" + name + "'",
                     ProjectDiagnosticErrorCode.MISSING_PKG_INFO_IN_BALLERINA_TOML.diagnosticId(),
                     DiagnosticSeverity.WARNING, pkgNode.location());
-            name = defaultName(this.projectPath).value();
         }
         version = getStringValueFromTomlTableNode(pkgNode, VERSION);
         if (version == null) {
-            reportDiagnostic("missing 'version' under [package]",
+            version = defaultVersion().value().toString();
+            reportDiagnostic("missing 'version' under [package], value derived as '" + version + "'",
```

Comment:
`missing key 'version' in table '[package]' in 'Ballerina.toml'. Defaulting to 'version = 0.1.0'`

Reply:
Refactored via https://github.com/ballerina-platform/ballerina-lang/pull/36174/commits/48fba38888323c8a03144c32f77884571964360f